storage/operation/multi_reader.go

```diff
@@ -0,0 +1,81 @@
+package operation
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/utils/noop"
+)
+
+type multiReader struct {
+	readers []storage.Reader
+}
+
+var _ storage.Reader = (*multiReader)(nil)
+
+// NewMultiReader returns a Reader that consists of multiple readers
+// in the provided order.  Readers are read sequentially until
+// - a reader succeeds or
+// - a reader returns an error that is not ErrNotFound
+// If all readers return ErrNotFound, Reader.Get will return ErrNotFound.
+func NewMultiReader(readers ...storage.Reader) storage.Reader {
+	if len(readers) == 1 {
+		return readers[0]
+	}
+	return &multiReader{readers: readers}
```

Comment:
Let's also ensure `len(readers)` is at least 1, because otherwise the `Get` method will always return exception, and could let the exception returned as early as from the constructor.